### ISSUE TITLE
Concurrent requests and bugs on --max-jobs

### ISSUE THREAD

**Issue by Porkechebure** [sd-webui/stable-diffusion-webui#1041]
### What happened?

Regarding bug #669 (please look at the screenshots in it).
The bug persists. You can't do any multitasking work and --max jobs doesn't do anything.

I need it to do multiple work like this:
![image](https://user-images.githubusercontent.com/3466040/189560355-f6c46815-725a-46ec-8839-9e49932558e8.png)

Moreover, the --max-jobs is bugged.

If you specify max jobs parameter you get an "incorrect int value"
to fix, i had to add a space before the parameters specified later in relauncher:
![image](https://user-images.githubusercontent.com/3466040/189560481-dabba585-584c-43a2-8561-899d1199fab2.png)

Please fix both the things

### Version

0.0.1 (Default)

### What browsers are you seeing the problem on?

Chrome

### Where are you running the webui?

Windows

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by outlookhazy** [comment]
See also #858 
I can't comment on request interleaving specifically which you talk about in #669, but without job-manager concurrent requests from multiple users/browsers work - they just show "loading" until they're done (no status updates)

**Comment by codedealer** [comment]
Job manager seems to be built with "one user only" approach in mind. I suggest running webui with `--no-job-manager` if you need concurrency for now.

**Comment by Porkechebure** [comment]
> Job manager seems to be built with "one user only" approach in mind. I suggest running webui with `--no-job-manager` if you need concurrency for now.

Gonna try and update to see if it solves.
In any case the thing is totally broken. Even if I am the only person doing it, it still does jobs 1 at time, so it's definitely not doing what it's supposed to.

**Comment by cobryan05** [comment]
--max-jobs should allow for multiple web browser tabs... I have made a few PRs to update JobManager which keep getting lost: (https://github.com/sd-webui/stable-diffusion-webui/pull/611 https://github.com/sd-webui/stable-diffusion-webui/pull/796) but I don't think they should be affecting session management.

What do you have max-jobs set to? You should be able to run one browser tab per job. What happens in the second browser tab when you hit generate?

**Comment by outlookhazy** [comment]
no idea what version of anything is being used since it's automatically installed, how would I check? 
Check the discussion on #858, 2 sessions work for the first request, then never again without restarting. Queuing and multiple sessions works fine without job-manager, so I'm not inclined to blame gradio

**Comment by cobryan05** [comment]
That should be gradio 3.1.6.... you can verify by loading checking gradio.__version__ in the Python interpreter

```
Microsoft Windows [Version 10.0.19044.1889]
(c) Microsoft Corporation. All rights reserved.

 F:\projects\stable-diffusion>C:/ProgramData/Miniconda3/Scripts/activate

(base) F:\projects\stable-diffusion>conda activate ldm

(ldm) F:\projects\stable-diffusion>python
Python 3.8.5 (default, Sep  3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import gradio
>>> gradio.__version__
'3.1.6'
```

How many jobs are you running? (Are your screenshots at the top your actual config? max-jobs=4?)
What environment are you in? (Windows, Linux, Docker...)
Does the JobManager status text box say anything?

Here's my computer running 3 jobs simultaneously:
![image](https://user-images.githubusercontent.com/13701027/189930684-9d7abeba-9011-415d-9ca2-c68b589c381c.png)

so without more information I won't be able to help much, because everything is working for me :-/

**Comment by outlookhazy** [comment]
Not OP, but I'm personally running under Docker and my post has exactly how to duplicate, including the commit hash. Admittedly, I haven't re-tried in a couple days, but I also didn't see anything in the commit history mentioning a change to job-manager. I haven't changed max-jobs from whatever the default is, my _assumption_ is that would mean simultaneous jobs, which would need to be 1 on my little 8G card. I can't handle simultaneous requests, that would be the point of the queue. In any case, the system becomes _completely_ unresponsive, to _any_ user as indicated in that report.

**Comment by cobryan05** [comment]
See if making this typo fix change addresses your issue:
https://github.com/sd-webui/stable-diffusion-webui/pull/1086/files

**Comment by outlookhazy** [comment]
lol, yep - that fixed the deadlock

**Comment by outlookhazy** [comment]
now if only "number of images" were done as a self-continuation rather than a single job. I'd love to be able to have 50 variations generating for one user and still be able to submit 1 in the middle from a different session that runs in the middle of the batch instead of after all the others are done - guess that'll be the next feature request

**Comment by cobryan05** [comment]
I'm actually locally working on a model manager that would allow that, but no promises on anything ever coming of it.

You might try just increasing max-jobs to 2 and see if it works on your device.  I'm not sure exactly what all changes have been pulled into this repo, but there have been some significant memory fixes out there.
Running from my repo https://github.com/cobryan05/stable-diffusion-webui/tree/dev  on my 1080ti  (11GB VRAM) I am able to run 1 job with a batch size of ~18, but generally just run batch sizes 2-3 but max-jobs=5.

TL;DR: if your 8GB card couldn't support more than 1 job a month ago, that doesn't necessarily mean it can't support multiple jobs now

**Comment by outlookhazy** [comment]
batch size of even 2 fails, though I haven't turned on any optimization that isn't default in the docker image

**Comment by codedealer** [comment]
You might try turning optimization on and giving it a go. I have 1080ti same as @cobryan05, with `--optimized` I could run `--batch_size 100` on the current `dev` branch.

**Comment by cobryan05** [comment]
Ah, it doesn't look like the memory fixes have made it to this repo yet. But at least there is hope for you running more on that card :) I'm sure they'll make it here soon enough. There is a PR for it (I think I cherry-picked an earlier version of this PR https://github.com/sd-webui/stable-diffusion-webui/pull/974 into my repo https://github.com/sd-webui/stable-diffusion-webui/commit/13be19cdf1bac673e54e7c269da59bbe3e087a11 ) , so hopefully they get their issues sorted out

**Comment by outlookhazy** [comment]
How much of a performance/quality hit is the optimized version? Documentation is very lacking with what the implications actually are (it's not even actually listed under the command line wiki page, and on its own tab it isn't explained other than where it might _not_ work)

**Comment by cobryan05** [comment]
--max_jobs=2 *should* allow two conccurent users.  It sets gradio's  concurrency_count to max_jobs+1  (to allow 1 thread for button presses).

I've never tried 'shared' mode, but it looks like it skips setting the max jobs on gradio:

```
        if not opt.share:
            demo.queue(concurrency_count=opt.max_jobs)
        if opt.share and opt.share_password:
            gradio_params['auth'] = ('webui', opt.share_password)
```

are you running with shared=True?

**Comment by outlookhazy** [comment]
No, I'm using a paired WireGuard container to share it and accessing it via 7860 on a static IP without shared mode
with max_jobs=2 it a set of images for 1 user, then the other user. bumping it up to 5 and they're interleaved (I can see 2 status bars at a time and refresh on each session shows partially completed sets as one would expect)

**Comment by cobryan05** [comment]
I don't see an update to gradio anywhere, but that is the behavior that new Gradio has, and max-jobs=2 is working for me fine on gradio==3.1.6.

Newer version their concurrency queue just doesn't seem to count concurrency correctly and will queue up jobs when there *should* be free jobs.

But whatever, no matter the version of gradio this should fix it: https://github.com/sd-webui/stable-diffusion-webui/pull/1092  It just tells JobManager related functions to not use gradio's queue.

**Comment by cobryan05** [comment]
> now if only "number of images" were done as a self-continuation rather than a single job. I'd love to be able to have 50 variations generating for one user and still be able to submit 1 in the middle from a different session that runs in the middle of the batch instead of after all the others are done - guess that'll be the next feature request
@outlookhazy - https://github.com/sd-webui/stable-diffusion-webui/pull/1361 should allow this. It's a bit WIP, but it seems to function at the moment
